### PR TITLE
Feature/infp 2431

### PR DESCRIPTION
**Pull request by dungnh165877**

### CODE REVIEW

File: api/v1/dto/hep.go [hunk 1]
```diff
@@ -32,36 +34,45 @@ type CreateHostEndpointInput struct {
...
 }
 
 type HostEndpointSpecInput struct {
 	InterfaceName string   `json:"interfaceName" yaml:"interfaceName"`
+	TenantID      uint64   `json:"tenantID" yaml:"tenantID" validate:"omitempty"`
+	IP            string   `json:"ip" yaml:"ip" validate:"omitempty,ip"`
 	IPs           []string `json:"ips" yaml:"ips" validate:"min=1,unique,dive,ip"`
 }
 
+type ListHEPsInput struct {
```
Comment (quantonganh) [inline api/v1/dto/hep.go]: I would use "ListHostEndpointsInput".

File: api/v1/mapper/gns.go [hunk 2]
```diff
@@ -6,6 +6,14 @@ import (
 	"github.com/bamboo-firewall/be/domain/model"
 )
 
+func ToListGlobalNetworkSetDTOs(gnss []*entity.GlobalNetworkSet) []*dto.GlobalNetworkSet {
+	var gnsDTOs []*dto.GlobalNetworkSet
```
Comment (quantonganh) [inline api/v1/mapper/gns.go]: You can pre-allocate len/cap by using `len(gnss)`.

File: api/v1/mapper/hep.go [hunk 3]
```diff
@@ -3,9 +3,18 @@ package mapper
 import (
 	"github.com/bamboo-firewall/be/api/v1/dto"
 	"github.com/bamboo-firewall/be/cmd/server/pkg/entity"
+	"github.com/bamboo-firewall/be/cmd/server/pkg/net"
 	"github.com/bamboo-firewall/be/domain/model"
 )
 
+func ToListHostEndpointDTOs(heps []*entity.HostEndpoint) []*dto.HostEndpoint {
+	var hepDTOs []*dto.HostEndpoint
```
Comment (quantonganh) [inline api/v1/mapper/hep.go]: same here.

File: cmd/bamboofwcli/command/delete.go [hunk 4]
```diff
@@ -13,25 +13,46 @@ import (
...
+  # Delete many policy with name
+  bbfw delete gnp allow_ssh allow_ping
+
+  # Delete many policy with filename
+  bbfw delete gnp -f allow_ssh.yaml -f allow_ping.yaml
+
+  # Delete a set with name
+  bbfw delete gns server
 
   # Delete many policy with name
```
Comment (quantonganh) [inline cmd/bamboofwcli/command/delete.go]: "many sets".

File: cmd/bamboofwcli/command/delete.go [hunk 5]
```diff
@@ -13,25 +13,46 @@ import (
...
-  bbfwcli delete hep allow_ssh allow_ping
+  bbfw delete gns server vm
 
   # Delete many policy with filename
```
Comment (quantonganh) [inline cmd/bamboofwcli/command/delete.go]: "many sets".

File: cmd/bamboofwcli/command/delete.go [hunk 6]
```diff
@@ -118,7 +153,7 @@ func deleteResources(cmd *cobra.Command, args []string) error {
 	for _, r := range resources {
 		err = resourceMgr.Delete(context.Background(), apiServer, r.Content)
 		if err != nil {
-			fmt.Printf("fail to delete resource from: %v\n", err)
+			fmt.Printf("fail to delete resource %s from: %v\n", r.Name, err)
```
Comment (quantonganh) [inline cmd/bamboofwcli/command/delete.go:156]: use `%w` to wrap error.

File: cmd/bamboofwcli/command/list.go [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+	ListHEPsByTenantID uint64
+	ListHEPsByIP       string
+
+	ListGNPsByIsOrder bool
+)
+
+var listCMD = &cobra.Command{
+	Use:   "list",
+	Short: "List resource",
+	Example: `  # List global network policy
```
Comment (quantonganh) [inline cmd/bamboofwcli/command/list.go]: "global network sets".

File: cmd/bamboofwcli/command/list.go [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+		input = &dto.ListGNPsInput{IsOrder: ListGNPsByIsOrder}
+	default:
+		return fmt.Errorf("unsupported resources type: %s", resourceType)
+	}
+
+	apiServer := client.NewAPIServer(os.Getenv(common.APIServerENV))
+
+	resources, err := resourceMgr.List(context.Background(), apiServer, input)
+	if err != nil {
+		return fmt.Errorf("list resources failed: %v", err)
```
Comment (quantonganh) [inline cmd/bamboofwcli/command/list.go]: You should use `%w` to wrap error.

File: cmd/server/pkg/entity/common.go [hunk 9]
```diff
@@ -1,17 +1,28 @@
...
 
 const (
-	ProtocolTCP  Protocol = "tcp"
-	ProtocolUDP  Protocol = "udp"
-	ProtocolICMP Protocol = "icmp"
-	ProtocolSCTP Protocol = "sctp"
+	ProtocolTCP     = "tcp"
+	ProtocolUDP     = "udp"
+	ProtocolICMP    = "icmp"
+	ProtocolSCTP    = "sctp"
```
Comment (quantonganh) [inline cmd/server/pkg/entity/common.go:23]: first time I saw this. Please add a comment for it.

File: cmd/server/pkg/repository/gnp.go [hunk 10]
```diff
@@ -44,19 +97,25 @@ func (r *PolicyDB) DeleteGNPByName(ctx context.Context, name string) *ierror.Cor
...
+		return errlist.ErrDatabase.WithChild(fmt.Errorf("delete global network policy failed: %w", err))
 	}
 	return nil
 }
 
-func (r *PolicyDB) ListGNP(ctx context.Context) ([]*entity.GlobalNetworkPolicy, *ierror.CoreError) {
+func (r *PolicyDB) ListGNPs(ctx context.Context, input *model.ListGNPsInput) ([]*entity.GlobalNetworkPolicy, *ierror.CoreError) {
+	var opts []*options.FindOptions
+	if input != nil {
+		if input.IsOrder {
```
Comment (quantonganh) [inline cmd/server/pkg/repository/gnp.go:108]: `if input != nil && input.IsOrder`.

File: cmd/bamboofwcli/command/delete.go (hunk 6 above)
Comment (dungnh165877) [reply cmd/bamboofwcli/command/delete.go:156]: This is fmt.Printf not fmt.Errorf ạ

**Comment by quantonganh** [comment]
LGTM.

**Pull request merged**